drivers/modem/ublox-sara-r4.c

```diff
@@ -687,6 +819,42 @@ static int pin_init(void)
 	return 0;
 }
 
+#if defined(CONFIG_MODEM_UBLOX_SARA_AUTODETECT_VARIANT)
+static void modem_rssi_query_work(struct k_work *work)
+{
+	static struct modem_cmd cmds[] = {
+		  MODEM_CMD("+CSQ: ", on_cmd_atcmdinfo_rssi_csq, 2U, ","),
+		  MODEM_CMD("+CESQ: ", on_cmd_atcmdinfo_rssi_cesq, 6U, ","),
+	};
+	static char *send_cmd_u2 = "AT+CSQ";
+	static char *send_cmd_r4 = "AT+CESQ";
+	int ret;
+
+	/* choose cmd according to variant */
+	char *send_cmd = send_cmd_r4;
+
+	if (mdata.mdm_variant == MDM_VARIANT_UBLOX_U2) {
+		send_cmd = send_cmd_u2;
+	}
+
+	/* query modem RSSI */
+	ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+						 cmds, ARRAY_SIZE(cmds),
+						 send_cmd,
+						 &mdata.sem_response,
+						 MDM_CMD_TIMEOUT);
+	if (ret < 0) {
+		LOG_ERR("AT+C[E]SQ ret:%d", ret);
```

Comment:
If we have an error, should we bail out here?

Reply:
we don't bail out here, because we want to restart a new RSSI query, regardless of teh outcome of this one. This is also how it is done in the original code.

Reply:
not really - we want to start the next query afterwards, regardless the outcome of this one.